core/server/models/plugins/collision.js

```diff
@@ -0,0 +1,82 @@
+var moment = require('moment-timezone'),
+    Promise = require('bluebird'),
+    _ = require('lodash'),
+    errors = require('../../errors');
+
+module.exports = function (Bookshelf) {
+    var ParentModel = Bookshelf.Model,
+        Model;
+
+    Model = Bookshelf.Model.extend({
+        /**
+         * Collision protection.
+         *
+         * NOTE: The `sync` method is called for any query e.g. update, add, delete, fetch
+         *
+         * NOTE: We had the option to override Bookshelf's `save` method, but hooking into the `sync` method gives us
+         *       the ability to access the `changed` object. Bookshelf already knows which attributes has changed.
+         *
+         * NOTE: Bookshelf's timestamp function can't be overridden, as it's synchronous, there is no way to return an Error.
+         *
+         */
+        sync: function timestamp(options) {
+            var parentSync = ParentModel.prototype.sync.apply(this, arguments),
+                originalUpdateSync = parentSync.update,
+                self = this, err;
+
+            if (this.tableName !== 'posts' ||
+                !self.serverData ||
+                ((options.method !== 'update' && options.method !== 'patch') || !options.method)
+            ) {
+                return parentSync;
+            }
+
+            /**
+             * Only hook into the update sync
+             *
+             * IMPORTANT NOTES:
+             * Even if the client sends a different `id` property, it get's ignored by bookshelf.
+             * Because you can't change the `id` of an existing post.
+             *
+             * HTML is always auto generated, ignore.
+             * @TODO: how to detect TAGS?
+             */
+            parentSync.update = function update(attrs) {
+                var changed = _.omit(self.changed, [
+                        'created_at', 'updated_at', 'author_id', 'id', 'tags',
+                        'published_by', 'updated_by', 'html'
+                    ]),
+                    clientUpdatedAt = moment(self.clientData.updated_at || self.serverData.updated_at),
+                    serverUpdatedAt = moment(self.serverData.updated_at);
+
+                if (Object.keys(changed).length) {
+                    if (clientUpdatedAt.diff(serverUpdatedAt) !== 0) {
```

Comment:
In Ghost-Admin we don't modify the `updated_at` property at all so it should always match what was received from the server.

I can't think of any use cases where we'd want to set the `updated_at` client-side unless we want to explore that avenue when we have offline support - e.g. we want the `updated_at` to reflect when the save request was made in the client rather than when it was eventually saved on the server - that seems like a remote edge case and maybe not the best idea anyway.